packages/docusaurus-theme-classic/src/utils/index.ts

```diff
@@ -7,6 +7,6 @@
 
 // Compare the 2 paths, ignoring trailing /
 export const isSamePath = (path1, path2) => {
-  const normalize = (str) => (str.endsWith('/') ? str : `${str}/`);
-  return normalize(path1) === normalize(path2);
+  const normalize = (str) => (str?.endsWith('/') ? str : `${str}/`);
```

Comment:
This is another safeguard for undefined. With such a fix it will work even without filtering done above and lead to a consistent Navbar menu structure.